src/app/pages/rpc-sign-psbt/rpc-sign-psbt.tsx

```diff
@@ -78,14 +56,24 @@ function useRpcSignPsbt() {
           },
         })
       );
+      window.close();
     },
   };
 }
 
 export function RpcSignPsbt() {
-  const { origin, decodedPsbt, onSignPsbt, onCancel } = useRpcSignPsbt();
+  const { allowedSighash, origin, decodedPsbt, indexesToSign, onSignPsbt, onCancel, tx } =
+    useRpcSignPsbt();
   if (!decodedPsbt) return null;
   return (
-    <PsbtSigner appName={origin} onSignPsbt={onSignPsbt} onCancel={onCancel} psbt={decodedPsbt} />
+    <PsbtSigner
+      allowedSighash={allowedSighash}
+      indexesToSign={indexesToSign}
+      origin={origin}
+      onSignPsbt={onSignPsbt}
+      onCancel={onCancel}
+      psbtRaw={decodedPsbt}
+      psbtTx={tx}
```

Comment:
Do we need to pass in both? One can be derived from the other, no?

Reply:
They are both derived in different getters from the passed psbt hex. They aren't derived from each other. So, I can just pass psbt hex itself and get what I need later if clearer?